07_safety/reported-crime/reported-crime-place-all.Rmd

```diff
@@ -0,0 +1,540 @@
+      violent = ifelse(ucr_offense_code %in% violent, 1, 0)
+    ) %>%
+    # Aggregate property and violent crimes by agency
+    group_by(ori) %>%
+    summarize(all = n(),
+              violent = sum(violent),
+              property = sum(property)) %>%
+    ungroup() %>%
+    # Create variable for year
+    mutate(year = as.character(year))
+}
+
+```
+
+### 3.3 Read in (and clean) NIBRS offense data
+```{r}
+# Note that future updates may want to include historical data and should un-comment the lines for 2020 and before
+offenses_2022 <- clean_offense_data(year=2022)
+offenses_2021 <- clean_offense_data(year=2021)
+# offenses_2020 <- clean_offense_data(year=2020)
+# offenses_2019 <- clean_offense_data(year=2019)
+# offenses_2018 <- clean_offense_data(year=2018)
+# offenses_2017 <- clean_offense_data(year=2017)
+# offenses_2016 <- clean_offense_data(year=2016)
+
+```
+
+### 3.4 Append NIBRS offense data for all years
+```{r}
+# Note that future updates may want to include historical data and should use the lines that are currently commented out
+offenses_agency <- bind_rows(
+  offenses_2021, offenses_2022
+)
+
+rm(codes, violent, property, clean_offense_data,
+   offenses_2021, offenses_2022)
+
+# offenses <- bind_rows(
+#   offenses_2014, offenses_2015, offenses_2016, offenses_2017, offenses_2018,
+#   offenses_2019, offenses_2020, offenses_2021, offenses_2022
+# )
+# 
+# rm(codes, violent, property, clean_offense_data,
+#    offenses_2014, offenses_2015, offenses_2016, offenses_2017, offenses_2018, 
+#    offenses_2019, offenses_2020, offenses_2021, offenses_2022)
+
+```
+
+Note that because the process of extracting and reading in these files takes so long, I write out the appended all arrests data here. Then I read this file back in when returning to this program so that I don't have to constantly re-run the previous steps.
+```{r}
+# Write out agency-level offense data
+write_csv(offenses_agency, file = here::here("07_safety", "reported-crime", "data", "offenses_21_22.csv"))
+
+# Read agency-level offense data back in for ease of future use
+offenses_agency <- read.csv(here::here("07_safety", "reported-crime", "data", "offenses_21_22.csv"))
+
+```
+
+
+## 4. Aggregate agency-level offenses to the place-level
+This section aggregates the agency-level offense counts (all, violent, and property) from the agency-level to the place-level.
+
+Our data are currently long by year/agency. However, we want to know the arrest counts at the place level, not the agency level. In the `01_agency_geo_place.Rmd` program, we constructed a file that linked reporting agencies to places. We can use that file to aggregate counts from agencies to places.
+
+### 4.1 Load universe of place/agency pairs
+First load the universe of agencies linked to places. This file was created in `01_agency_geo_place.Rmd` and is long by unique agency/place pairs.
+```{r}
+bhf_place_urb <- read_csv(here::here("07_safety", "modified data", "all_agency_place.csv"))
```

Comment:
Thank you for catching this!!! I *knew* something seemed wrong with this but I could not figure out what it was. I applied this change to `bhf_place_urb` in `01_agency_geo_place` instead so that I don't have to change it here AND in `juvenile-arrests-place-all`.